modules/http/http.c

```diff
@@ -306,14 +306,17 @@ http_dd_init(LogPipe *s)
   HTTPDestinationDriver *self = (HTTPDestinationDriver *)s;
   GlobalConfig *cfg = log_pipe_get_config(s);
 
+  if (self->load_balancer->num_targets == 0)
+    http_load_balancer_add_target(self->load_balancer, HTTP_DEFAULT_URL);
+
+  /* we need to set up url before we call the inherited init method, so our stats key is correct */
+  self->url = self->load_balancer->targets[0].url;
```

Comment:
Does this (using `targets[0]` as `url`) mean that when the user prepend a new target to the target list and reloads syslog-ng, we lost the diskq files?

Reply:
I have added a warning for this case, advising the user to set persist-name()